### PR TITLE
added time and date to error report

### PR DESCRIPTION
**Pull request by J3fftw1**
## Description

added time and date to the error report, this makes it easier for us to see when this error has been thrown. instead of just blindly trusted the user

## Proposed changes

Added time and date to the error report

## Related Issues (if applicable)

N/A

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/api/ErrorReport.java [hunk 1]
```diff
@@ -84,6 +86,10 @@ public ErrorReport(T throwable, SlimefunAddon addon, Consumer<PrintStream> print
     @ParametersAreNonnullByDefault
     public ErrorReport(T throwable, Location l, SlimefunItem item) {
         this(throwable, item.getAddon(), stream -> {
+            Date date = new Date();
+            SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/api/ErrorReport.java:90]: This should not be initialised every single error report. Instead make this a global static variable

File: src/main/java/io/github/thebusybiscuit/slimefun4/api/ErrorReport.java [hunk 2]
```diff
@@ -84,6 +86,10 @@ public ErrorReport(T throwable, SlimefunAddon addon, Consumer<PrintStream> print
...
+
+            stream.println(formatter.format(date));
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/api/ErrorReport.java:92]:
```suggestion
            stream.println(formatter.format(new Date()));
```

**Review by WalshyDev — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/api/ErrorReport.java [hunk 3]
```diff
@@ -3,9 +3,11 @@
 import java.io.File;
 import java.io.PrintStream;
 import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
 import java.util.ArrayList;
+import java.util.Date;
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/api/ErrorReport.java:10]: unused import

File: src/main/java/io/github/thebusybiscuit/slimefun4/api/ErrorReport.java [hunk 4]
```diff
@@ -84,6 +86,8 @@ public ErrorReport(T throwable, SlimefunAddon addon, Consumer<PrintStream> print
     @ParametersAreNonnullByDefault
     public ErrorReport(T throwable, Location l, SlimefunItem item) {
         this(throwable, item.getAddon(), stream -> {
+
+            stream.println("Error Generated: " + dateFormat.format(LocalDateTime.now()));
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/api/ErrorReport.java:90]: This should not be location-based but rather added to **every** error-report in my opinion. So please move that down to the part where Java version etc... are added as well

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Review by TheBusyBiscuit — APPROVED** [review]
The wording could be improved but looks fine to me

**Review by WalshyDev — APPROVED** [review]

**Pull request merged**